R/promise.R

```diff
@@ -465,16 +465,17 @@ as.promise.Future <- function(x) {
         future::resolved(x, timeout = 0)
       }, FutureError = function(e) {
          reject(e)
-         TRUE
+         NULL
       })
+      if (is.null(is_resolved)) return()
```

Comment:
By returning `NULL`, we can quick execution within the promise. 

Before the PR, it would continue to ask for the `future::value()`. Even if one was found, the `resolve(result)` would be a no-op as `reject(e)` has already been called.

Therefore, we should quit early